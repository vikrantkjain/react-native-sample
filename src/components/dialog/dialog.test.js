import React from 'react';
import Dialog from './index';
import {render, screen, fireEvent} from '@testing-library/react-native';
const mockReanimated = jest.requireMock('react-native-reanimated');
const mockCam = jest.requireMock('react-native-vision-camera');
jest.mock('@react-navigation/bottom-tabs', () => ({
  createBottomTabNavigator: jest.fn(),
}));
jest.mock('react-native-reanimated', () => {
  return {
    ...mockReanimated,
    useSharedValue: jest.fn,
    useAnimatedStyle: jest.fn,
  };
});
jest.mock('react-native-image-crop-picker', () => 'ImagePicker');
jest.mock('react-native-simple-toast', () => 'Toast');
jest.mock('react-native-config', () => 'RNConfig');
jest.mock('@react-native-async-storage/async-storage', () => 'AsyncStorage');
jest.mock('react-native-sensitive-info', () => 'RNSInfo');
jest.mock('@react-native-community/netinfo', () => 'NetInfo');
jest.mock('react-native-linear-gradient', () => 'LinearGradient');
jest.mock('react-native-vision-camera', () => {
  return {
    ...mockCam,
    useCameraDevices: jest.fn,
  };
});

const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};

describe('Dialog render correctly with testing lib', () => {
  const onPressMock = jest.fn(() => null);
  const tree = render(
    <Dialog
      visible={true}
      animationType="fade"
      okPress={onPressMock}
      title="Test title"
    />,
  );
  it('Renders successfully', () => {
    expect(tree).toBeDefined();
  });
  it('renders correctly', () => {
    const treeAsJSON = tree.toJSON();
    expect(treeAsJSON).toMatchSnapshot();
  });
  fireEvent.press(screen.getByTestId('TestButton-testID'), eventData);
  expect(onPressMock).toHaveBeenCalledWith(eventData);
});
