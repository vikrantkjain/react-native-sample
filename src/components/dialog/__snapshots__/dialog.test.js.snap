// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog render correctly with testing lib renders correctly 1`] = `null`;
